test/parallel/test-os.js

```diff
@@ -101,3 +102,15 @@ switch (platform) {
 
 var EOL = os.EOL;
 assert.ok(EOL.length > 0);
+
+
+var home = os.homedir();
+
+console.log('homedir = ' + home);
+assert.ok(home.length > 0);
+
+if (isWindows && process.env.USERPROFILE) {
+  assert.equal(home, process.env.USERPROFILE);
+} else if (!isWindows && process.env.HOME) {
+  assert.equal(home, process.env.HOME);
+}
```

Comment:
Maybe as an extra check delete HOME / USERPROFILE from the env and check that `os.homedir()` still returns something.